runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -782,8 +787,13 @@ private void checkVersion() {
                             getVersionString(), version.getVersion());
                 }
             }
-        } catch (TimeoutException | NetworkException | ShutdownException e) {
-            log.error("connect: failed to get version", e);
+        } catch (TimeoutException | NetworkException | ShutdownException | UnavailableServerException e) {
```

Comment:
Why do you catch Exception? aren't the new exceptions coming from adding UnavailableServerException?

Reply:
Since the checkVersion is purely informational, regardless of whatever exception occurs, if it passes or not, no further action is taken... If we don't catch any generic Exception we could be bubbling up other type of exceptions we are not aware of, and that are irrelevant to this step. In this case the separation was only meant to further provide detailed log info, but at this point we care of not throwing up any kind of exception...

Reply:
checkVersion is being called only from connect() on Runtime. Even if we are sure that important exceptions as ShutdownException or UnavailabeServerException are again during execution of connect(), would it be better re-throw them here in checkVersion so that connect() deals with them?